server/api/utils/singletons/k8s.py

```diff
@@ -533,7 +533,7 @@ def store_secrets(
                     )
                 raise exc
 
-        secret_data = k8s_secret.data.copy()
+        secret_data = (k8s_secret.data or {}).copy()
```

Comment:
you don't need to copy an empty dict, so how about this?
```suggestion
        secret_data = k8s_secret.data.copy() of k8s_secret.data else {}
```

Reply:
You right about no need to copy an empty dict but I think the way I did it is more compact.

Reply:
yes it's compact, but you created 2 objects instead of one.